app/listeners/rummager_notifier.rb

```diff
@@ -1,19 +1,17 @@
 class RummagerNotifier
-  def self.notify(changed_query_match_type_pairs)
-    changed_query_match_type_pairs.each do |(query_string, match_type)|
-      update_elasticsearch(query_string, match_type)
+  def self.notify(queries_with_action)
+    queries_with_action.each do |query, action|
+      update_elasticsearch(query, action)
     end
   end
 
-  def self.update_elasticsearch(query_string, match_type)
-    query = Query.where(query: query_string, match_type: match_type).first
-
-    if query && query.bets.any?
+  def self.update_elasticsearch(query, action)
+    if action == :delete || (action == :update_bets && query.bets.empty?)
```

Comment:
As discussed this needs a larger refactor as the indirection being used here is not valuable.